File: Data Structures & Algorithms/two-integer-sum/submission-7.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        # [3:1, 4:2, 5:3, 6:4]
        
        tar_finder = {}
        for i in range(len(nums)):
            print(tar_finder)
            j = target - nums[i]
            if j in tar_finder:
                return [tar_finder[j], i]

            tar_finder[nums[i]] = i

        return [0, 1] 

File: Data Structures & Algorithms/two-integer-sum-ii/submission-7.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // sort
        sort(numbers.begin(), numbers.end());

        int lp = 0;
        int rp = numbers.size() - 1;
        while (lp < rp){
            if (numbers[lp] + numbers[rp] == target) {
                return {lp + 1, rp + 1};
            } 
            else if (numbers[lp] + numbers[rp] > target) {
                rp--;
            }
            else {
                lp++;
            }
        }
        return {};
    }
};
